app/views/foreman_bootdisk/generic_efi_host.erb

```diff
@@ -0,0 +1,59 @@
+#
+# Boot disk Grub2 EFI - generic host
+#
+#set debug="http,efinet,net"
+#set debug=all
+echo "Foreman Bootdisk: 'Boot disk Grub2 EFI - generic host' template"
+echo
+echo "********************************************************"
+echo " REQUIREMENTS:"
+echo " * SUBNET GENERIC IMAGE ONLY (host image not supported)"
+echo " * HOST PARAM default_grub_install_entry set to efi_http"
+echo " * PROXY WITH HTTPBOOT FEATURE"
+echo " * HTTP UEFI BOOT ONLY (Legacy/PXE not supported)"
+echo " * IPv4 ONLY (IPv6 not tested, change the template)"
+echo " * HTTP ONLY (change the template for HTTPS)"
+echo " * ISC DHCP (other servers not tested)"
+echo " * GRUB FROM RHEL 8.3+/7.9+ (when generating the image)"
+echo " * EFI HTTP or HTTPS grub entry must be selected in menu"
+echo "*******************************************************"
+sleep 5
+<%
+# possible values are: "http" or "https"
+proxy_proto = "http"
+
+@subnet || bootdisk_raise("Generic disk not supported for EFI, use subnet disk")
+@subnet.httpboot? || bootdisk_raise("Requires a proxy with httpboot feature")
+@subnet.template? || bootdisk_raise("Requires a proxy with template feature")
+proxy_port = @subnet.httpboot.setting(:HTTPBoot, "#{proxy_proto}_port") || bootdisk_raise("Requires proxy with #{proxy_proto}_port exposed setting")
+proxy_httpboot_host = @subnet.httpboot.hostname
+proxy_template_host = @subnet.template.hostname
```

Comment:
Don't we need to resolve the hostnames to ipaddresses the same way as we do here https://github.com/theforeman/community-templates/blob/f2f58d2c67ce8d47fde09f78f91142682403c224/provisioning_templates/PXEGrub2/kickstart_default_pxegrub2.erb#L39 ?

Reply:
We absolutely do, my smart proxy had IP address since I was testing this on devel setup.